### PR TITLE
Indicate engine version in docs and API

### PR DESCRIPTION
**Pull request by MichaelMraka**
engine version is shown in metrics as
```
# HELP patchman_engine_core_info Patchman project deployment information
# TYPE patchman_engine_core_info gauge
patchman_engine_core_info{version="v1.4.2"} 1

```

### CODE REVIEW

File: base/metrics/metrics.go [hunk 1]
```diff
@@ -15,10 +15,23 @@ var (
 	}, []string{"type"})
 )
 
+var (
```
Comment (Josca) [inline base/metrics/metrics.go:18]: We can join these two `var` sections into one.

File: base/metrics/metrics.go [hunk 2]
```diff
@@ -15,10 +15,23 @@ var (
...
+)
+
+var ENGINEVERSION = "v1.4.1"
+
 func init() {
 	if os.Getenv("KAFKA_ADDRESS") != "" {
 		prometheus.MustRegister(KafkaConnectionErrorCnt)
 	}
+	prometheus.MustRegister(EngineVersion)
+	EngineVersion.With(prometheus.Labels{"version":ENGINEVERSION}).Set(1)
```
Comment (Josca) [inline base/metrics/metrics.go]: Good, but `EngineVersion.WithLabelValues(ENGINEVERSION).Set(1)` is easier.

File: scripts/generate_docs.sh [hunk 3]
```diff
@@ -1,5 +1,25 @@
 #!/usr/bin/bash
 
+RELEASE=""
```
Comment (Josca) [inline scripts/generate_docs.sh:3]: Could you describe how to use this script with a new options in the heading? I mean e.g.:
~~~bash
# ./generate_docs.sh --increment # Increment "z" in version
# ./generate_docs.sh --keep # Keep current version
~~~

File: scripts/generate_docs.sh [hunk 4]
```diff
@@ -1,5 +1,25 @@
...
+case "$1" in
+  -i|--increment)
+        CURRENT_RELEASE=$(git tag | tail -n1)
+        RELEASE="${CURRENT_RELEASE%.*}.$((${CURRENT_RELEASE##*.}+1))"
+        ;;
+  -k|--keep) # don't change anything, just keep current release
+        ;;
+  -r|--release) RELEASE=$2
```
Comment (Josca) [inline scripts/generate_docs.sh:11]: Cool. What to add `--version` or `--get-version` to see current version and then stop?

File: scripts/generate_docs.sh [hunk 5]
```diff
@@ -1,5 +1,25 @@
...
+  *) >&2 echo "Usage: $0 [ [-k|--keep] | [-i|--increment] | [-r|--release] <release> ]"
+        exit 1
+        ;;
+esac
+
+if [ -n "$RELEASE" ] ; then
+  # Substitute version
+  sed -i "s|\(// @version \).*$|\1 $RELEASE|;" manager/manager.go
+  sed -i 's|^\(var ENGINEVERSION = "\)[^"]*\("\)$|'"\1$RELEASE\2|;" base/metrics/metrics.go
+fi
```
Comment (Josca) [inline scripts/generate_docs.sh:22]: It's possible someone will edit only manually only one place where version is filled (doc var or go var). What to add checking that these constants are equal?

**Review by Josca — CHANGES_REQUESTED** [review]
Looks good, just a few notes how to improve that. Thanks.

File: base/metrics/metrics.go (hunk 1 above)
Comment (MichaelMraka) [reply base/metrics/metrics.go]: fixed

File: base/metrics/metrics.go (hunk 2 above)
Comment (MichaelMraka) [reply base/metrics/metrics.go]: fixed

File: scripts/generate_docs.sh (hunk 3 above)
Comment (MichaelMraka) [reply scripts/generate_docs.sh:13]: added

File: scripts/generate_docs.sh (hunk 4 above)
Comment (MichaelMraka) [reply scripts/generate_docs.sh:24]: added

File: scripts/generate_docs.sh (hunk 5 above)
Comment (MichaelMraka) [reply scripts/generate_docs.sh:44]: generate_docs.sh --get-version will show if the version differ
I've also added a warning comment to not edit this value

**Review by Josca — APPROVED** [review]
great, thanks

**Pull request merged**